nagios/test/test_unit.py

```diff
@@ -0,0 +1,56 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from mock import patch
+
+from datadog_checks.nagios import Nagios
+from .common import CHECK_NAME, CUSTOM_TAGS
+
+# Random test values
+METRIC_NAME = 'nagios.metric_name'
+METRIC_VALUE = 42
+METRIC_TAGS = CUSTOM_TAGS
+METRIC_TIMESTAMP = 1337404007
+
+
+class TestGaugeWrapper:
+
+    def test_gauge_without_timestamp(self):
+        """
+        Test the 'gauge' wrapper to see if it strips the 'timestamp' arg for gauge functions that do not accept it
+        """
+
+        def timestamp_excluded_gauge(self, name, value, tags=None):
+            """
+            A gauge function that does not accept the 'timestamp' argument
```

Comment:
I'm not sure if there's a better way to test this, I previously allowed this function to receive `**kwargs`, but it got messy when it came time to assert. This way, however, there'll be an exception that gets raised if this is called with a `timestamp`.